Kitodo/src/main/java/org/kitodo/production/helper/SearchResultGeneration.java

```diff
@@ -60,40 +55,19 @@ public SXSSFWorkbook getResult() {
         return getWorkbook();
     }
 
-    private List<ProcessDTO> getResultsWithFilter() {
-        List<ProcessDTO> processDTOS = new ArrayList<>();
-        try {
-            processDTOS = ServiceManager.getProcessService().findByQuery(getQueryForFilter(ObjectType.PROCESS),
-                ServiceManager.getProcessService().sortById(SortOrder.ASC), true);
-        } catch (DataException e) {
-            logger.error(e.getMessage(), e);
+    private List<Process> getResultsWithFilter() {
+        BeanQuery query = new BeanQuery(Process.class);
+        if (StringUtils.isNotBlank(filter)) {
+            query.restrictWithUserFilterString(filter);
         }
-
-        return processDTOS;
-    }
-
-    /**
-     * Gets the query with filters.
-     *
-     * @param objectType Type of object that should be filtered
-     * @return A BoolQueryBuilder
-     */
-    public BoolQueryBuilder getQueryForFilter(ObjectType objectType) {
-        BoolQueryBuilder query = new BoolQueryBuilder();
-
-        try {
-            query = ServiceManager.getFilterService().queryBuilder(this.filter, objectType, false, false);
-        } catch (DataException e) {
-            logger.error(e.getMessage(), e);
-        }
-
         if (!this.showClosedProcesses) {
-            query.mustNot(ServiceManager.getProcessService().getQueryForClosedProcesses());
+            query.restrictToNotCompletedProcesses();
         }
         if (!this.showInactiveProjects) {
-            query.mustNot(ServiceManager.getProcessService().getQueryProjectActive(false));
+            query.addBooleanRestriction("project.active", Boolean.FALSE);
```

Comment:
```suggestion
            query.addBooleanRestriction("project.active", Boolean.TRUE);
```

You are defaulting to Boolean.FALSE, which breaks Excel file generation because only processes of inactive projects are included.

Reply:
Should this not depend on the current used filter? There you can be defining if you want to see disabled projects or not and depending on this setting the excel list generation should include them or not? Or did I miss something as I do not use this excel list generation feature on a regular base.

Reply:
The logic is a little confusing here, but i think what is meant is the following:
- if the user has NOT selected to show/include INACTIVE projects, we need to apply a filter to our search. That filter is: SELECT ONLY ACTIVE projects
- if he has selected to to show/include INACTIVE projects, we do no filtering at as we select all the projects.
- What is wrong here is, that when the user has NOT selected to show/include INACTIVE projects, only INACTIVE projects are selected, while correct would be to select only ACTIVE projects.

Reply:
Thank you for the explanation @BartChris 

In this way it is more understandable and should be adjusted.

Reply:
Maybe we can just add a comment for clarity:

```java
  //If 'showClosedProcesses' is not selected we want to restrict the the selection to unfinished processes.
   if (!this.showClosedProcesses) {
            query.restrictToNotCompletedProcesses();
        }

   // If 'showInactiveProjects' is not selected we want to restrict the included processes to only those belonging to active projects.
    if (!showInactiveProjects) {
        query.addBooleanRestriction("project.active", Boolean.TRUE);
    }
```

Reply:
True, I was confused by all the negatives. If it's not supposed to show inactive projects, it obviously has to limit it to active ones.